pkg/actions/addon/update.go

```diff
@@ -77,7 +77,7 @@ func (a *Manager) Update(ctx context.Context, addon *api.Addon, podIdentityIAMUp
 	}
 
 	var deleteServiceAccountIAMResources []string
-	if len(summary.PodIdentityAssociations) > 0 && !addon.UseDefaultPodIdentityAssociations {
+	if len(summary.PodIdentityAssociations) > 0 && !addon.UseDefaultPodIdentityAssociations && !a.clusterConfig.AddonsConfig.AutoApplyPodIdentityAssociations {
 		if addon.PodIdentityAssociations == nil {
```

Comment:
qq/non-blocking: What's the reason behind using an array pointer here? (`PodIdentityAssociations *[]PodIdentityAssociation`

Reply:
To remove pod identity associations from an addon, a user needs to explicitly set this field to the empty slice. A pointer is used to distinguish between an unspecified field and an empty slice.